### PR TITLE
Fix clippy lints for Rust 1.54

### PR DESCRIPTION
**Pull request by PoignardAzur**
Fixes the CI errors that pop up on new pull requests since Rust 1.54 was released.

### CODE REVIEW

**Comment by PoignardAzur** [comment]
@cmyr Assuming this passes CI, I intend to merge it in 24h if nobody has objected in the meantime.

File: druid-shell/src/lib.rs [hunk 1]
```diff
@@ -26,7 +26,7 @@
 //! - `DRUID_SHELL_DISABLE_X11_PRESENT`: if this is set and `druid-shell` is using the `x11`
 //! backend, it will avoid using the Present extension.
 
-#![warn(broken_intra_doc_links)]
+#![warn(rustdoc::broken_intra_doc_links)]
```
Comment (Maan2003) [inline druid-shell/src/lib.rs:29]: does not compile on 1.51 :/

Comment (PoignardAzur) [reply druid-shell/src/lib.rs:29]: ... That's annoying. Fine, I'll just revert it.

Comment (Maan2003) [reply druid-shell/src/lib.rs:29]: add a `// TODO(MSRV)` :)

**Review by cmyr — APPROVED** [review]
Thanks!

**Pull request merged**